Sumar tarea

Iara-Pou opened the pull request:
Muestro screens de como se vería la interfaz: 

**EJERCICIO 1**
![image](https://user-images.githubusercontent.com/92182454/201341369-6bdcadeb-74d6-42c3-bdb5-201cfaec20d7.png)
![image](https://user-images.githubusercontent.com/92182454/201341240-627cd3a4-2b89-4481-95fb-e92db430131f.png)
![image](https://user-images.githubusercontent.com/92182454/201341282-0dd01bcc-4756-4092-8a59-177d4ef7c787.png)

**EJERCICIO 2**
![image](https://user-images.githubusercontent.com/92182454/201341594-6e73ad75-994e-43c3-8555-012ac23f9b4c.png)
![image](https://user-images.githubusercontent.com/92182454/201341475-b318438e-9176-48e2-9d8b-276afd9e872c.png)
![image](https://user-images.githubusercontent.com/92182454/201341622-a4d0634e-b4a1-4ab0-b001-0c0685791bf8.png)

Iara-Pou committed:
Suma ejercicios hasta ahora

Iara-Pou committed:
Completa atributo en label

No estaba referenciando a nada el label, lo linkee al input.

Iara-Pou committed:
Incorporo NPM

Modulo de bootstrap, más el link.

Iara-Pou committed:
Saca el cambio de letra

Iara-Pou committed:
Suma link al módulo de bootsrap para ejercicio 2

Iara-Pou committed:
Crea grid

Iara-Pou committed:
Arregla bug

No se estaba mostrando correctamente el width de los botones calcular y reiniciar porque al ejecutar la funcion esconder botones y mostrar botones, se eliminaban o reemplazaban todas las clases y no solo la clase oculto.

Iara-Pou committed:
Saca estilos innecesarios

Iara-Pou committed:
Suma etiqueta semántica

Iara-Pou committed:
Suma estilos con bootstrap

Iara-Pou committed:
Borra css innecesario

Iara-Pou committed:
Arregla bugs con classList

Antes no me mostraban las clases aplicadas de bootstrap porque esconder errores las borraba con el classList.

Iara-Pou committed:
Cambia estilos

Iara-Pou committed:
Arregla funcionamiento botón cantidad

Primero, agrega vaciarErrores porque se estaban acumulando los errores en el cartel. después cambia la forma de sacar la clase oculto para no eliminar las de bootstrap.

Iara-Pou committed:
Suma estilos a cada input

Se crean con classList en cada iteración, según la cantidad de integrantes de la familia.

Iara-Pou committed:
Arregla bug

Pasaba lo mismo con el input que tiene la cantidad de integrantes.

Iara-Pou committed:
Suma estilos bootstrap

Iara-Pou committed:
Agrega texto para cada input con su error

Para que sea más específico, itera cada input y les asigna su error con un elemento p. al alert que avisa que hubo un error, le pone un mensaje general.

Iara-Pou committed:
Centra el cartel de errores

Iara-Pou committed:
Arregla bug

El texto individual de cada error no se borraba una vez que este se haya resuelto.

Iara-Pou committed:
Arregla bug errores edades

Se acumulaba un error abajo del otro, por lo que seleccioné todo el texto por debajo de los inputs y lo borré antes de sumar texto nuevo.

Iara-Pou committed:
Arregla funcionalidad del botón reinicio

Antes reiniciaba todos los inputs porque era type reset pero no tenía sentido en contexto. así que le cambié el tipo y cree otra función onclick que se encargue de vaciar los inputs creados y no el primero (para eso estaba el otro botón de reinicio).

Iara-Pou committed:
Suma etiquetas semánticas

Iara-Pou committed:
Cambia link al css de bootstrap comprimido

Iara-Pou committed:
Suma botón remover al lado de cada input

Para poder seleccionar que salario en específico quiero eliminar, era más práctico ponerlo al lado de cada input y que no sea un botón general.

Iara-Pou committed:
Suma función al botón remover

Agrega la función como un parámetro en la función que lo construye, para eliminar su elemento padre.

Iara-Pou committed:
Elimina el botón restar integrante

Ya que no cumple ninguna función que no cumplan los botones individuales.

Iara-Pou committed:
Cambia elemento label por p

Dado que no tiene un input vinculado, no tenía sentido que sea un label.

Iara-Pou committed:
Arregla bug

Cuando eliminaba todos los inputs, seguía mostrandose el botón calcular. Entonces, hice que si el contenedor estuviera vacío, se esconda al hacer click en el botón remover.

Iara-Pou committed:
Suma grid de bootstrap

Iara-Pou committed:
Arregla bug botones remover

Antes, cuando hacía el cálculo correctamente se seguían mostrando los botones remover, lo que causaba que se puedan eliminar el input.
Por eso les sumé una clase, para que puedan ser seleccionados y escondidos.

Iara-Pou committed:
Arregla bug botones remover

Si apretaba sumar integrante no podía eliminar inputs anteriores, porque se mantenían escondidos los botones de remover input.

Iara-Pou committed:
Arregla bug

Si antes habían ingresado un input con error y se mostró en el cartel de errores, para después eliminar todos los integrantes, se seguía viendo el error en la pantalla.
Por eso, si ahora el contenedor está vacío, también se borran los errores anteriores.

Iara-Pou committed:
Suma clases bootstrap con js

Iara-Pou committed:
Da estilo a botones remover

Antes eran muy altos y estaban pegados al margen de arriba, entonces los corregí con una clase de CSS.

Iara-Pou committed:
Suma clases bootstrap al index

Para dar estilo.

Iara-Pou committed:
Arregla bug

Se mostraba desde el principio el cartel con los errores (vacío), entonces creé una función para eso.

Iara-Pou committed:
Elimina estilo innecesario

Fue reemplazado por un alert.

Iara-Pou committed:
Saca tags strong innecesarios

Iara-Pou committed:
Arregla bug

Cuando se ejecutaba correctamente el programa, se seguía viendo el cartel.
 Por eso, suma función esconderCartelErrores.

Iara-Pou committed:
Corrige alineación de elementos

Iara-Pou committed:
Arregla bug en errores

Estaba borrando todos los errores anteriores al último en la iteración, ya que se repetía por iteración. Entonces, lo moví a la función vaciar errores anteriores.

Iara-Pou committed:
Resume código

Iara-Pou committed:
Crea funciones mostrar y esconder

Para evitar la repetición del código
"elemento..classList.remove("oculto");" o "elemento.classList.add("oculto");", cree dos funciones que agregen o quiten la clase oculto de los elementos.

Iara-Pou committed:
Corrige código

Cambié el nombre $mensaje por $mensajeResultado y el nombre $contenedor por $contenedorInputs para más claridad de lectura, además de formatearlo.

Iara-Pou committed:
Update pruebas.js

Iara-Pou committed:
Borra código cubierto por mostrar() y esconder()

Iara-Pou committed:
Cambia archivo cálculos

Ahora le sumé las funciones mostrar, esconder y conseguirInputPorValor, que se comparten en ambos ejercicios. Le cambié el nombre a funciones compartidas porque era más descriptivo.

Iara-Pou committed:
Mueve funciones compartidas a carpeta general

Para que se pueda acceder más fácil para los dos archivos.

Iara-Pou committed:
Linkea y organiza scripts

Por el archivo funcionesCompartidas, para que no repitan código innecesario.

Iara-Pou committed:
Cambia ancho de labels (con input)

Porque se veía muy desorganizado para usarlo.

Iara-Pou committed:
Ordena el script

Iara-Pou commented on clase-6-tarea/ejercicio-2/js/script.js:
```diff
@@ -0,0 +1,145 @@
+    botonRemover.textContent = `Eliminar integrante`;
+    botonRemover.classList = "boton-remover btn btn-danger col-sm-3 ";
+
+    botonRemover.onclick = function () {
```
No sé si es muy desordenado agregar el event handler adentro de la creación del botón, ¿Sería mejor crear una función nueva y asignarselo al onclick?

Genebson replied:
Podes hacer eso o podes crear el botón directamente en el html, y luego esconderlo/mostrarlo dependiendo del workflow de la tarea. Es decir, cuando crean integrantes, aparece el botón de remover y así.

Iara-Pou replied:
Claro, no sabía como poner uno al lado de cada input de esa forma. Sería uno general para todos los inputs si hago eso, no? Porque no se sabe cuantas veces el usuario puede apretar el botón sumar integrante 🤔

Genebson replied:
Ahí deberías pensar la lógica de tu aplicación. Cómo querés que funcione? Digamos, el usuario pone 4 en el input que genera los integrantes => qué pasa acá? El botón que genera integrantes, sigue habilitado o lo deshabilitas?. Si lo deshabilitas, agregas un botón que resetee. Si lo dejas activo, tenes que buscar la forma en la que si creó 4 integrantes y luego quiere agregar 2 más, esos 2 últimos tiene que ser integrante 5 e integrante 6

Iara-Pou commented on clase-6-tarea/funcionesCompartidas.js:
```diff
@@ -0,0 +1,46 @@
+    return numeroMayor;
+}
+
+function mostrar(elemento) {
```
Estoy dejando mucha variable global y es mejor que vuelva a crear funciones de a una para esconder y mostrar elementos, ¿O así está bien? Lo hice de esta manera porque tenía un montón de lineas repetidas que decían `elemento.classList.add("oculto")` o `elemento.classList.remove("oculto")`

Genebson replied:
Cuales serían las variables globales? 🤔

Iara-Pou replied:
```
const $contenedorErrores = document.querySelector("#errores");
const $mensajeResultado = document.querySelector("#calculo");
```
Esas dos, antes estaban adentro de funciones siempre.

Genebson replied:
Esas dos variables en específico me parece que está bien que sean globales si las reutilizas en varias funciones

Iara-Pou replied:
Okeey, lo dejo así entonces, genial!

Iara-Pou commented on clase-6-tarea/funcionesCompartidas.js:
```diff
@@ -0,0 +1,46 @@
+function devolverPromedio(numeros) {
```
Una duda más por la ubicación del archivo, asumí que así está bien porque es más fácil de acceder desde ambos scripts, pero no sé si puede ser que sea muy desordenado para proyectos de mayor escala. 
¿Es mala práctica?

Genebson replied:
Al contrario, es una buena práctica separar por archivos. Cuando te toque hacer la pokedex vas a aprender a separar tu proyecto por módulos, es decir, las funcionalidades de la ui en una carpeta que se llame ui, el servicio, en su carpeta de servicios y así...

Iara-Pou committed:
Suma salto de línea

Iara-Pou committed:
Arregla bug

Se seguía mostrando el mensaje con los errores después de hacer el reinicio general.

Iara-Pou committed:
Merge branch 'sumar-tarea' of https://github.com/Iara-Pou/tarea-clase-10 into sumar-tarea

Genebson commented on clase-6-tarea/ejercicio-2/js/script.js:
```diff
@@ -0,0 +1,145 @@
+    elementoPadre.appendChild(botonRemover);
+}
+
+function guardarSalariosEnArray(inputs) {
```
Acá no hace falta aclarar donde guardas los salarios, directamente podes poner ``guardarSalarios``.

Iara-Pou replied:
Ya lo cambié, gracias!

Genebson commented on clase-6-tarea/ejercicio-2/js/script.js:
```diff
@@ -0,0 +1,145 @@
+}
+
+function guardarSalariosEnArray(inputs) {
+    const array = [];
```
Lo mismo acá: ``const salarios = [];``

Genebson commented on clase-6-tarea/ejercicio-2/js/script.js:
```diff
@@ -0,0 +1,145 @@
+}
+
+function crearBotonRemover(elementoPadre) {
+    let botonRemover = document.createElement("button");
```
Intentá no aclarar el elemento en la variable, si el día de mañana deja de ser un botón y pasa a ser un div tenes que cambiarlo por todos lados. Puede ir con ``const`` esa variable me parece

Iara-Pou committed:
Cambia nombres en guardarSalariosEnArray

Ahora la función se llama guardarSalarios y la constante que devuelve se llama salarios, para que sea más claro en el ejemplo.

Iara-Pou committed:
Crea un parámetro para crearBotonRemover

Como puede ser un botón o un div, cree un parámetro que por default sea button, para que siga funcionando en su contexto, pero que se cubra en caso de que sea necesario un div u otro elemento en su lugar.

Iara-Pou committed:
Crea función removerContenedor

Creé la función para no acumular el código de que hace en la misma función que lo crea.

Genebson commented on clase-6-tarea/ejercicio-1/js/script.js:
```diff
@@ -0,0 +1,161 @@
+function guardarEdadesIntegrantes(numeros) {
+    const arrayNuevo = [];
```
Esta variable puede llamarse edades

Genebson commented on clase-6-tarea/ejercicio-1/js/script.js:
```diff
@@ -0,0 +1,161 @@
+    $contenedorInputs.innerHTML = "";
+}
+
+function esconderBotones() {
```
Yo esta función la fletaría y dejaría solo esconder y mostrar

Iara-Pou committed:
Cambia nombre de variable en guardarEdadesIntegrantes

En lugar de poner arrayNuevo, escribí edades, para que sea más específico de la función.

Iara-Pou committed:
Borra mostrarBotones y esconderBotones

Las reemplaza por mostrar y esconder.
